Document/prepare serde deserialization better?

Hi there,

even following code examples from the docs, I struggled a bit getting `into_json()` to work for me, running into these errors and warning:
```
error: cannot find derive macro `Deserialize` in this scope
 --> src/main.rs:4:10
  |
4 | #[derive(Deserialize)]
  |          ^^^^^^^^^^^

warning: unused import: `serde::Deserialize`
 --> src/main.rs:2:5
  |
2 | use serde::Deserialize;
  |     ^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0277]: the trait bound `for<'de> JsonData: Deserialize<'de>` is not satisfied
  --> src/main.rs:20:42
   |
20 |     let data: JsonData = request.into_json()?;
   |                                          ^^^^^^^^^ the trait `for<'de> Deserialize<'de>` is not implemented for `JsonData`
   |
   = note: required because of the requirements on the impl of `DeserializeOwned` for `JsonData`
```

I was able to solve this by not only adding serde to my project's `Cargo.toml`, but also specify the derivate feature:
```toml
[…]

[dependencies]
serde = { version = "1", features = ["derive"] }
ureq = { version = "2.2.0", features = ["json"] }
```

Can this be documented better?

Or, can this even be made unnecessary by changing ureq's own `Cargo.toml` like this (I'm no dependecy expert)?
```diff
[…]

[dependencies]
-serde = { version = "1", optional = true }
+serde = { version = "1", optional = true, features = ["derive"] }
serde_json = { version = "1", optional = true }
```

--- a/src/response.rs
+++ b/src/response.rs
@@ -376,7 +376,13 @@ impl Response {
     /// ```
     /// # fn main() -> Result<(), ureq::Error> {
     /// # ureq::is_test(true);
-    /// use serde::{Deserialize, de::DeserializeOwned};
+    /// // This import requires the `derive` feature on `serde`.
+    /// // Put this in Cargo.toml: serde = { version = "1", features = ["derive"] }
+    /// use serde::{Deserialize};
+    ///
+    /// // An alternative way to is to use the `serde_derive` crate.
+    /// // Put this in Cargo.toml: serde_derive = "1"
+    /// // use serde_derive::{Deserialize};
     ///
     /// #[derive(Deserialize)]
     /// struct Message {

